core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.component.ts

```diff
@@ -134,12 +134,42 @@ export class TopicsAndSkillsDashboardPageComponent {
     this.directiveSubscriptions.unsubscribe();
   }
 
-  generateNumbersTillRange(range: number): number[] {
-    let arr: number[] = [];
-    for (let i = 0; i < range; i++) {
-      arr.push(i);
+  getPages(current: number, total: number): number[] {
+    /**
+     * This function returns an array of the total number of pages.
+     * "-1" is an indicator to print ellipses.
+     * The "total" variable is being rounded off to avoid cases
+     * when the items per page is not a divisor of the total
+     * number of pages.
+     */
+    total = Math.ceil(total);
+    let initialPaginationThreshold: number = 6;
+    /**
+     * Case 0 when the total count of pages is less than or equal
```

Comment:
When the total count of ... to 6, ellipses aren't required.

(note: arent --> aren't)